Sources/Commands/SwiftTool.swift

```diff
@@ -608,12 +608,7 @@ public class SwiftTool {
         } else {
             // User didn't tell us to use these .netrc files so be more lenient with errors
             func loadNetrcNoThrows(at path: AbsolutePath) -> NetrcAuthorizationProvider? {
-                guard localFileSystem.exists(path) else { return nil }
-                
-                do {
-                    try withTemporaryFile(dir: path.parentDirectory) { _ in }
-                } catch {
-                    self.observabilityScope.emit(warning: "\(path.parentDirectory) is not accessible or not writable, not using .netrc file in it: \(error)")
+                guard localFileSystem.exists(path) && localFileSystem.isReadable(path) else {
```

Comment:
`addOrUpdate` is outside the `AuthorizationProvider` protocol iirc, so theoretically you would need to cast into `NetrcAuthorizationProvider` to use it?

Reply:
Yes, `addOrUpdate` is not part of the `AuthorizationProvider` protocol currently, but I see potential of it being added later (e.g., for registry authn).

Reply:
I am hesitant about it requiring write access to this one specifically since often its a readonly file

we should def consider that if we are to offer a "write" netrc utility. atm i dont think we do, we can also comment out that code with a comment to make it more clear?

Reply:
> we should def consider that if we are to offer a "write" netrc utility

We would have to. For saving registry credentials.

Reply:
One option might be to modify `NetrcAuthorizationProvider.addOrUpdate` so it does the `isWritable` check each time, and then we keep this `isReadable`.

Reply:
@yim-lee happy to add such check if you think its useful. lmk!